crates/rslint_core/src/groups/errors/no_dupe_class_members.rs

```diff
@@ -0,0 +1,230 @@
+        It can cause unexpected behaviors.
+
+        ## Incorrect code examples
+
+        ```js
+        class Foo {
+            bar() { }
+            bar() { }
+        }
+
+        class Foo {
+            bar() { }
+            get bar() { }
+        }
+
+        class Foo {
+            static bar() { }
+            static bar() { }
+        }
+
+        ```
+
+        ## Correct code examples
+
+        ```js
+        class Foo {
+            bar() { }
+            qux() { }
+        }
+
+        class Foo {
+            get bar() { }
+            set bar(value) { }
+        }
+
+        class Foo {
+            static bar() { }
+            bar() { }
+        }
+        ```
+    */
+    #[derive(Default)]
+    NoDupeClassMembers,
+    errors,
+    "no-dupe-class-members"
+}
+
+#[typetag::serde]
+impl CstRule for NoDupeClassMembers {
+    fn check_node(&self, node: &SyntaxNode, mut ctx: &mut RuleCtx) -> Option<()> {
+        if node.kind() == CLASS_BODY {
+            let getters = node
+                .children()
+                .filter_map(|n| {
+                    if n.kind() == GETTER {
+                        Some(n.to::<ast::Getter>())
+                    } else {
+                        None
+                    }
+                })
+                .collect();
+            if let Some(d) = check(self.name(), &mut ctx, getters) {
+                ctx.add_err(d);
+                return None;
+            }
+
+            let setters = node
+                .children()
+                .filter_map(|n| {
+                    if n.kind() == SETTER {
+                        Some(n.to::<ast::Setter>())
+                    } else {
+                        None
+                    }
+                })
+                .collect();
+            if let Some(d) = check(self.name(), &mut ctx, setters) {
+                ctx.add_err(d);
+                return None;
+            }
+
+            let methods = node
+                .children()
+                .filter_map(|n| {
+                    if n.kind() == METHOD {
+                        Some(n.to::<ast::Method>())
+                    } else {
+                        None
+                    }
+                })
+                .collect();
+            if let Some(d) = check(self.name(), &mut ctx, methods) {
+                ctx.add_err(d);
+                return None;
+            }
+        }
+
+        None
+    }
+}
+
+fn check<T: Named>(code: &str, ctx: &mut RuleCtx, nodes: Vec<T>) -> Option<Diagnostic> {
+    let mut names = HashMap::new();
+    for n in &nodes {
+        let mut name = Named::name(n);
+        if name == "constructor" {
+            continue;
+        }
+        if n.is_static() {
+            name = format!("static {}", name);
+        }
+        if let Some(prev_decl) = names.insert(name.clone(), n.name_node()) {
+            let err = ctx
+                .err(code, format!("Duplicate name '{}'", name))
```

Comment:
```suggestion
                .err(code, format!("Duplicate name `{}`", name))
```
We use backticks for code/names in the project usually, so this should be consistent too